docs/src/design/stack/io_ops.md

```diff
@@ -0,0 +1,193 @@
+$$
+
+In the above:
+- $op_{mem}$ is the unique [operation label](../chiplets/main.md#operation-labels) of the memory access operation.
+- $s_0$ is the memory address from which the value is to be loaded onto the stack.
+- $clk$ is the current clock cycle of the VM.
+
+The effect of this operation on the rest of the stack is:
+* **No change** starting from position $1$.
+
+### MSTOREW
+Assume that the words with elements $v_0, v_1, v_2, v_3$ is located in memory at address $a$. The `MSTOREW` operation pops an element off the stack, interprets it as a memory address, and writes the remaining $4$ elements at the top of the stack into memory at the specified address. The stored elements are not removed from the stack. The diagram below illustrates this graphically.
+
+![mstorew](../../assets/design/stack/io_ops/MSTOREW.png)
+
+After the operation the contents of memory at address $a$ would be set to $u_0, u_1, u_2, u_3$.
+
+To simplify description of memory access request value, we first define the following variables:
+
+The value representing state of memory before the operation (set by the prover non-deterministically in registers $h_0, ..., h_3$):
+
+$$
+v_{old} = \sum_{i=0}^3\alpha_{i+4} \cdot h_i
+$$
+
+The value representing state of memory after the operation:
+
+$$
+v_{new} = \sum_{i=0}^3\alpha_{i+8} \cdot s_{3-i}'
+$$
+
+Using the above variables, we define the value representing the memory access request as follows:
+
+$$
+u_{mem} = \alpha_0 + \alpha_1 \cdot op_{mem} + \alpha_2 \cdot s_0 + \alpha_3 \cdot clk + v_{old} + v_{new}
+$$
+
+In the above:
+- $op_{mem}$ is the unique [operation label](../chiplets/main.md#operation-labels) of the memory access operation.
+- $s_0$ is the memory address into which the values from the stack are to be saved.
+- $clk$ is the current clock cycle of the VM.
+
+The effect of this operation on the rest of the stack is:
+* **Left shift** starting from position $1$.
+
+### MSTORE
+Assume that the words with elements $v_0, v_1, v_2, v_3$ is located in memory at address $a$. The `MSTORE` operation pops an element off the stack, interprets it as a memory address, and writes the remaining element at the top of the stack into the first element of the word located at the specified memory address. The remaining $3$ elements of the word are not affected. The diagram below illustrates this graphically.
+
+![mstore](../../assets/design/stack/io_ops/MSTORE.png)
+
+After the operation the contents of memory at address $a$ would be set to $b, v_1, v_2, v_3$.
+
+To simplify description of memory access request value, we first define the following variables:
+
+The value representing state of memory before the operation (set by the prover non-deterministically in registers $h_0, ..., h_3$):
+
+$$
+v_{old} = \sum_{i=0}^3\alpha_{i+4} \cdot h_i
+$$
+
+The value representing state of memory after the operation:
+
+$$
+v_{new} = \alpha_4 \cdot s_0' + \sum_{i=1}^3\alpha_{i+8} \cdot h_i'
+$$
```

Comment:
I think this should be $\alpha_8$ instead of $\alpha_4$

Reply:
Fixed!